shared/src/utils/sort-object.spec.ts

```diff
@@ -0,0 +1,55 @@
+import { cloneDeep } from 'lodash-es';
+import { sortObject } from './sort-object';
+
+const tests: SortObjectTest<any>[] = [
+    {
+        object: { c: 3, b: 2, a: 1 },
+        compareFn: ([keyA, valueA], [keyB, valueB]) => valueA - valueB,
+        expectedObject: { a: 1, b: 2, c: 3 },
+    },
+    {
+        object: { a: 1, b: 2, c: 3 },
+        compareFn: ([keyA, valueA], [keyB, valueB]) => valueA - valueB,
+        expectedObject: { a: 1, b: 2, c: 3 },
+    },
+    {
+        object: { c: 3, b: 2, a: 1 },
+        compareFn: ([keyA, valueA], [keyB, valueB]) => valueB - valueA,
+        expectedObject: { c: 3, b: 2, a: 1 },
+    },
+    {
+        object: { c: 3, b: 2, a: 1 },
+        compareFn: ([keyA, valueA], [keyB, valueB]) =>
+            (keyA as string).localeCompare(keyB as string),
+        expectedObject: { a: 1, b: 2, c: 3 },
+    },
+];
+
+describe('sortObject', () => {
+    it.each(tests)('$# sorts %j', ({ object, compareFn, expectedObject }) => {
+        const objectToBeSorted = cloneDeep(object);
+        const sortedObj = sortObject(objectToBeSorted, compareFn);
+        // The sorted object should be equal to the expected object
```

Comment:
Why aren't you using `toStrictEqual` directly?

Reply:
Because `toStrictEqual` doesn't check the order of the keys.

```ts
expect({ a: 1, b: 2 }).toStrictEqual({ b: 2, a: 1 }); // Pass
```

Reply:
Ah, good point. Right.